docs/visual-studio-extensibility/nuget-api-in-visual-studio.md

```diff
@@ -23,35 +41,72 @@ As of NuGet 3.3+, NuGet exports the following
 
-    When installed, the package automatically sets the **Embed Interop Types** property of the assembly reference to **True**. This makes your code  resilient against version changes when users update to newer versions of NuGet.
+## Using NuGet Services
 
 > [!Warning]
-> Do not use any other types besides the public interfaces in your code, and do not reference any other NuGet assemblies, including `NuGet.Core.dll`.
+> Do not use any other types besides the public interfaces in your code, and do not reference any other NuGet assemblies, such as `NuGet.Protocol.dll`, `NuGet.Frameworks.dll`, and so on.
+
+In order to maximize the backwards compatibility promises we make, but also providing ourselves the flexibility to implement new features, performance improvements, and bug fixes in Visual Studio, we do not support the NuGet Client SDK being used in Visual Studio, and we do not provide binding redirects in `devenv.exe.config` to assemblies other than our VS extensibility contracts.
+
+If you would like a new NuGet related API in Visual Studio, please search [NuGet's Home repo](https://github.com/NuGet/Home/) and up vote any existing issues if you find a similar one. If you can't find an existing feature request to up vote, please create one.
+
+### Brokered Services
+
+1. Install the [`NuGet.VisualStudio.Contracts`](https://www.nuget.org/packages/NuGet.VisualStudio.Contracts/) package into your project, as well as [`Microsoft.VisualStudio.SDK`](https://www.nuget.org/packages/Microsoft.VisualStudio.SDK).
+
+1. Use the `IAsyncServiceProvider` to get Visual Studio's service broker, and use that to get NuGet's service.
+
+   ```cs
+   // Your AsyncPackage implements IAsyncServiceProvider
+   IAsyncServiceProvider asyncServiceProvider = this;
+   var brokeredServiceContainer = await asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+   var serviceBroker = brokeredServiceContainer.GetFullAccessServiceBroker();
+   var nugetProjectService = await serviceBroker.GetProxyAsync<INuGetProjectService>(NuGetServices.NuGetProjectServiceV1);
+   ```
+
+1. When your code no longer needs NuGet's brokered service, dispose it. For example, if you only need NuGet's brokered service during a single method call, you can wrap it in a C#`using` statement:
+
+   ```cs
+   InstalledPackagesResult installedPackagesResult;
+   using (nugetProjectService as IDisposable)
+   {
+       installedPackagesResult = await nugetProjectService.GetInstalledPackages(projectGuid, cancellationToken);
+   }
+   ```
+
+
+### MEF Services
+
+1. Install the [`NuGet.VisualStudio`](https://www.nuget.org/packages/NuGet.VisualStudio) package into your project, which contains the `NuGet.VisualStudio.dll` assembly.
+
+    In NuGet 5.11 and earlier, the package automatically sets the **Embed Interop Types** property of the assembly reference to **True**. In NuGet 6.0, NuGet followed a change in Visual Studio guidance to stop using embed interop types.
```

Comment:
nit: phrasing, I'd say it's a Visual Studio change, rather than NuGet following a VS policy. 
So maybe say VS doesn't embed interop types in 2022 anymore.

Ideally link these?